src/containers/home/home.tsx

```diff
@@ -1,28 +1,43 @@
 import { Button, Stack, Typography } from "@mui/material";
-import React from "react";
+import React, { useState } from "react";
 import { signIn, useSession } from "next-auth/react";
 import { useRouter } from "next/router.js";
 
-import { text } from "~/assets/text/index.js";
+import { useRecaptcha } from "~/service/index.js";
 import { TopRanked } from "~/components/index.js";
+import { isValidSession } from "~/utils/session.js";
 import { api } from "~/utils/api.js";
+import { text } from "~/assets/text/index.js";
 import { pages } from "~/router.js";
 import { TOP_RANKED_PLAYERS } from "~/constants/index.js";
-import { isValidSession } from "~/utils/session.js";
-
 import { styles } from "./styles.js";
+import Script from "next/script";
 
 export const Homepage = () => {
   const { push } = useRouter();
+  const [isRecaptchaFailed, setIsRecaptchaFailed] = useState<boolean>(false);
+  const verifyCaptcha = api.recaptcha.verify.useMutation();
   const join = api.lobby.join.useMutation();
   const { data: sessionData } = useSession();
+  const { executeRecaptcha, id, src, onError, onLoad, strategy, ...props } =
```

Comment:
What about grouping `id`, `src`... and all the props you need for `Script` into one argument? It makes a bit cleaner import. then you can always references the props in the `Script` as `recaptchaProps.id`, `recaptchaProps.src`, etc.